### PR TITLE
Add fn init_emit()

### PR DESCRIPTION
**Pull request by wcampbell0x2a**
Grouped like initialize variables into one function that were copied between the emit functions for write/read

### CODE REVIEW

File: deku-derive/src/lib.rs [hunk 1]
```diff
@@ -274,6 +274,40 @@ impl DekuData {
     fn emit_writer_checked(&self) -> Result<TokenStream, syn::Error> {
         emit_deku_write(self)
     }
+
+    /// Create like initialize variables for emit write/read functions
+    fn init_emit(
```
Comment (sharksforarms) [inline deku-derive/src/lib.rs:279]: `emit_enum_common` might be a better name

File: deku-derive/src/lib.rs [hunk 2]
```diff
@@ -274,6 +274,40 @@ impl DekuData {
     fn emit_writer_checked(&self) -> Result<TokenStream, syn::Error> {
         emit_deku_write(self)
     }
+
+    /// Create like initialize variables for emit write/read functions
```
Comment (sharksforarms) [inline deku-derive/src/lib.rs:278]:
```suggestion
    /// Create common initializer variables for enum_emit read/write functions
```

File: deku-derive/src/lib.rs [hunk 3]
```diff
@@ -274,6 +274,40 @@ impl DekuData {
...
+    fn init_emit(
+        &self,
+    ) -> Result<
+        (
```
Comment (sharksforarms) [inline deku-derive/src/lib.rs:282]: Clippy error here, may be worth refactoring into a struct?

**Review by sharksforarms — CHANGES_REQUESTED** [review]
Couple small changes, looks good! :+1:

**Comment by sharksforarms** [comment]
Will need a rebase :)

**Comment by wcampbell0x2a** [comment]
- Refactored into struct, with a `try_from` from `DekuData` and then deconstruct into the variables.

File: deku-derive/src/lib.rs [hunk 4]
```diff
@@ -276,6 +277,51 @@ impl DekuData {
     }
 }
 
+/// Common variables from `DekuData` for enum_emit read/write functions
```
Comment (sharksforarms) [inline deku-derive/src/lib.rs:280]:
```suggestion
/// Common variables from `DekuData` for `emit_enum` read/write functions
```

File: deku-derive/src/lib.rs [hunk 5]
```diff
@@ -276,6 +277,51 @@ impl DekuData {
...
+    ident: TokenStream,
+    id: Option<&'a Id>,
+    id_type: Option<&'a TokenStream>,
+    id_args: TokenStream,
+}
+
+impl<'a> TryFrom<&'a DekuData> for DekuDataEnum<'a> {
+    type Error = syn::Error;
+
+    /// Create common initializer variables for enum_emit read/write functions
```
Comment (sharksforarms) [inline deku-derive/src/lib.rs:295]:
```suggestion
    /// Create common initializer variables for `emit_enum` read/write functions
```

File: deku-derive/src/macros/deku_read.rs [hunk 6]
```diff
@@ -140,30 +140,25 @@ fn emit_enum(input: &DekuData) -> Result<TokenStream, syn::Error> {
     let crate_ = super::get_crate_name();
     let mut tokens = TokenStream::new();
 
+    //let (imp, wher, variants, ident, id, id_type, id_args) = input.init_emit()?;
```
Comment (sharksforarms) [inline deku-derive/src/macros/deku_read.rs:143]: leftover comment

File: deku-derive/src/macros/deku_read.rs [hunk 7]
```diff
@@ -140,30 +140,25 @@ fn emit_enum(input: &DekuData) -> Result<TokenStream, syn::Error> {
...
+    let DekuDataEnum {
```
Comment (sharksforarms) [inline deku-derive/src/macros/deku_read.rs:144]: This look good :+1:

**Review by sharksforarms — COMMENTED** [review]
Thanks! The same could probably be done for emit_struct in read/write. Would you be interested in doing that too?

**Comment by wcampbell0x2a** [comment]
> Thanks! The same could probably be done for emit_struct in read/write. Would you be interested in doing that too?

sure!

**Comment by wcampbell0x2a** [comment]
Alright that should be good

**Review by sharksforarms — APPROVED** [review]
Thanks!

**Pull request merged**